### PR TITLE
Use Xcode 13 to build, iOS 15 unit test fixes

### PR DESCRIPTION
**Pull request by ewanharris**
This moves us to using Xcode 13 on Jenkins and GitHub, and fixes _some_ of the unit test failures on iOS 15.

Creating as a draft as the following util tests still fail and I need to investigate them:

1. #inspect() with nested object and infinite depth
2. #inspect() with object
3. #format() %o - object placeholder with same object twice
4. #format() %o - object placeholder with nested object
5. #format() %o - object placeholder with object

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -111,11 +111,12 @@ jobs:
         rm -f ~/.gradle/caches/modules-2/gc.properties
 
   ios:
-    runs-on: macos-latest
+    runs-on: macos-11
     name: iOS
     env:
       CCACHE_DIR: ${{ github.workspace }}/.ccache
       USE_CCACHE: 1
+      DEVELOPER_DIR: /Applications/Xcode_13.0.app/Contents/Developer
```
Comment (ewanharris) [inline .github/workflows/build.yml:119]: This isn't pretty, but it's the [only way](https://www.jessesquires.com/blog/2020/01/06/selecting-an-xcode-version-on-github-ci/)

File: Jenkinsfile [hunk 2]
```diff
@@ -128,7 +128,7 @@ def androidUnitTests(testName, nodeVersion, npmVersion, deviceId) {
 
 def macosUnitTests(nodeVersion, npmVersion) {
 	return {
-		node('git && xcode-12') {
+		node('git && xcode-13') {
```
Comment (ewanharris) [inline Jenkinsfile:131]: We currently only have 1 Xcode 13 node, so this build will take a little while. We'll roll Xcode 13 to another 2 soon

**Review by jquick-axway — APPROVED** [review]
LGTM

File: Jenkinsfile (hunk 2 above)
Comment (ewanharris) [reply Jenkinsfile:131]: We now have Xcode 13 on 3 nodes

File: tests/Resources/util.test.js [hunk 3]
```diff
@@ -489,6 +519,28 @@ describe('util', () => {
 						+ '  }\n'
 						+ '}'
 					);
+				} else if (OS_VERSION_MAJOR >= 15) {
+					// {\n  foo: 'bar',\n  foobar: 1,\n  func: <ref *1> [Function: func] {\n    [length]: 0,\n    [name]: 'func',\n    [prototype]: func { [constructor]: [Circular *1] }\n  }\n}
```
Comment (garymathews) [inline tests/Resources/util.test.js:523]: Maybe remove these comments since they match the condition below?

File: tests/Resources/util.test.js [hunk 4]
```diff
@@ -489,6 +519,28 @@ describe('util', () => {
...
+					// {\n  foo: 'bar',\n  foobar: 1,\n  func: <ref *1> [Function: func] {\n    [prototype]: func { [constructor]: [Circular *1] },\n    [name]: 'func',\n    [length]: 0\n  }\n}
```
Comment (garymathews) [inline tests/Resources/util.test.js:524]:
```suggestion
```

Comment (ewanharris) [reply tests/Resources/util.test.js:524]: Gah I thought I had deleted those, my bad!

**Pull request merged**